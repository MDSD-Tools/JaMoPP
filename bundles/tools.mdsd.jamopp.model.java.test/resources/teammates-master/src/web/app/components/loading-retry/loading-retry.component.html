<div *ngIf="shouldShowRetry; else content" class="text-center">
  <p class="text-muted m-1">{{message}}</p>
  <button (click)="retryEvent.emit()" class="btn btn-primary"><i class="fas fa-redo" aria-hidden="true"></i>&nbsp;Retry</button>
</div>

<ng-template #content>
  <ng-content></ng-content>
</ng-template>
