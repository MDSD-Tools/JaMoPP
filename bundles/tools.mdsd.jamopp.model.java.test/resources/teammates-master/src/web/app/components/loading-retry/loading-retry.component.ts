import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';

/**
 * Retry button for when loading fails.
 */
@Component({
  selector: 'tm-loading-retry',
  templateUrl: './loading-retry.component.html',
  styleUrls: ['./loading-retry.component.scss'],
})
export class LoadingRetryComponent implements OnInit {

  @Input() message: string = '';

  @Input() shouldShowRetry: boolean = false;

  @Output() retryEvent: EventEmitter<any> = new EventEmitter<any>();

  constructor() { }

  ngOnInit(): void {
  }

}
