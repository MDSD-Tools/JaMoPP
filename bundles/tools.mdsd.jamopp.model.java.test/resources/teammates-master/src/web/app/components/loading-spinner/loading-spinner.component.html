<div class="loading-container">
  <div class="row spinner-border text-primary" role="status"></div>
  <div class="row text-secondary">Loading...</div>
</div>
