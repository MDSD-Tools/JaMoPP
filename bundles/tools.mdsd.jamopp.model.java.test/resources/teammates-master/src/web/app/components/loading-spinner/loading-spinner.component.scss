.loading-container {
  margin: 1rem;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.spinner-border {
  width: 3rem;
  height: 3rem;
  display: inline-table;
}
