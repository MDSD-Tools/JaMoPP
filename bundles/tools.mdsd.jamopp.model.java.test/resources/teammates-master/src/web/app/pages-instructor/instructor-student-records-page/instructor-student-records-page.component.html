<h1 id="records-header">{{studentName}}'s Records<small class="text-muted text-break"> - {{courseId}}</small></h1>
<tm-loading-retry [shouldShowRetry]="hasStudentLoadingFailed || hasStudentProfileLoadingFailed" [message]="'Failed to load student records'" (retryEvent)="loadStudentRecords()">
  <tm-student-profile *tmIsLoading="isStudentProfileLoading" [studentProfile]="studentProfile" [photoUrl]="photoUrl"></tm-student-profile>
  <tm-more-info *tmIsLoading="isStudentLoading" [studentName]="studentName" [moreInfoText]="studentProfile.moreInfo"></tm-more-info>
</tm-loading-retry>
<h2>Records in feedback sessions</h2>
<tm-loading-retry [shouldShowRetry]="hasStudentResultsLoadingFailed" [message]="'Failed to load student results'" (retryEvent)="loadStudentResults()">
  <div *tmIsLoading="isStudentResultsLoading">
    <div *ngFor="let session of sessionTabs" class="card card-default mb-4">
      <div class="card-header cursor-pointer" (click)="session.isCollapsed = !session.isCollapsed">
        <strong class="text-break">Feedback Session: {{ session.feedbackSession.feedbackSessionName }}</strong>
        <div class="card-header-btn-toolbar">
          <tm-panel-chevron [isExpanded]="!session.isCollapsed"></tm-panel-chevron>
        </div>
      </div>
      <div class="card-body" *ngIf="!session.isCollapsed" @collapseAnim>
        <tm-grq-rgq-view-responses *ngIf="session.responsesReceivedByStudent.length > 0"
                                   [session]="session.feedbackSession"
                                   [responses]="session.responsesReceivedByStudent" [groupByTeam]="false" [section]="" [sectionType]=""
                                   [isGrq]="false" [instructorCommentTableModel]="instructorCommentTableModel"
                                   (saveNewCommentEvent)="saveNewComment($event, session.feedbackSession.timeZone)"
                                   (deleteCommentEvent)="deleteComment($event)"
                                   (updateCommentEvent)="updateComment($event, session.feedbackSession.timeZone)"
                                   [isExpandAll]="true">
        </tm-grq-rgq-view-responses>
        <div *ngIf="session.responsesReceivedByStudent.length === 0" class="mb-3">
          No feedback responses for {{studentName}} ({{studentTeam}}) found.
        </div>
        <tm-grq-rgq-view-responses *ngIf="session.responsesGivenByStudent.length > 0"
                                   [session]="session.feedbackSession"
                                   [responses]="session.responsesGivenByStudent" [groupByTeam]="false" [section]="" [sectionType]=""
                                   [isGrq]="true" [instructorCommentTableModel]="instructorCommentTableModel"
                                   (saveNewCommentEvent)="saveNewComment($event, session.feedbackSession.timeZone)"
                                   (deleteCommentEvent)="deleteComment($event)"
                                   (updateCommentEvent)="updateComment($event, session.feedbackSession.timeZone)"
                                   [isExpandAll]="true"
        ></tm-grq-rgq-view-responses>
        <div *ngIf="session.responsesGivenByStudent.length === 0">
          No feedback responses from {{studentName}} ({{studentTeam}}) found.
        </div>
      </div>
    </div>
  </div>
</tm-loading-retry>
