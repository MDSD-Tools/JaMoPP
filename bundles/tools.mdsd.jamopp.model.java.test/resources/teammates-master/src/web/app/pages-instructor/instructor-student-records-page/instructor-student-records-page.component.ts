import { Component, OnInit } from '@angular/core';
import { ActivatedRoute } from '@angular/router';
import { finalize, map, mergeMap } from 'rxjs/operators';
import { environment } from '../../../environments/environment';
import { FeedbackResponseCommentService } from '../../../services/feedback-response-comment.service';
import { FeedbackSessionsService } from '../../../services/feedback-sessions.service';
import { InstructorService } from '../../../services/instructor.service';
import { StatusMessageService } from '../../../services/status-message.service';
import { StudentProfileService } from '../../../services/student-profile.service';
import { StudentService } from '../../../services/student.service';
import { TableComparatorService } from '../../../services/table-comparator.service';
import {
  FeedbackSession,
  FeedbackSessions,
  Gender,
  Instructor,
  QuestionOutput,
  ResponseOutput,
  SessionResults,
  Student,
  StudentProfile,
} from '../../../types/api-output';
import { Intent } from '../../../types/api-request';
import { SortBy, SortOrder } from '../../../types/sort-properties';
import { CommentTableModel } from '../../components/comment-box/comment-table/comment-table.component';
import { CommentToCommentRowModelPipe } from '../../components/comment-box/comment-to-comment-row-model.pipe';
import { CommentsToCommentTableModelPipe } from '../../components/comment-box/comments-to-comment-table-model.pipe';
import { collapseAnim } from '../../components/teammates-common/collapse-anim';
import { ErrorMessageOutput } from '../../error-message-output';
import { InstructorCommentsComponent } from '../instructor-comments.component';

interface SessionTab {
  isCollapsed: boolean;
  feedbackSession: FeedbackSession;
  responsesGivenByStudent: QuestionOutput[];
  responsesReceivedByStudent: QuestionOutput[];
}

/**
 * Instructor student records page.
 */
@Component({
  selector: 'tm-instructor-student-records-page',
  templateUrl: './instructor-student-records-page.component.html',
  styleUrls: ['./instructor-student-records-page.component.scss'],
  animations: [collapseAnim],
})
export class InstructorStudentRecordsPageComponent extends InstructorCommentsComponent implements OnInit {

  courseId: string = '';
  studentName: string = '';
  studentEmail: string = '';
  studentTeam: string = '';
  studentSection: string = '';
  instructorCommentTableModel: Record<string, CommentTableModel> = {};

  studentProfile: StudentProfile = {
    name: '',
    shortName: '',
    email: '',
    institute: '',
    nationality: '',
    gender: Gender.OTHER,
    moreInfo: '',
  };
  sessionTabs: SessionTab[] = [];
  photoUrl: string = '';

  isStudentLoading: boolean = false;
  hasStudentLoadingFailed: boolean = false;
  isStudentProfileLoading: boolean = false;
  hasStudentProfileLoadingFailed: boolean = false;
  isStudentResultsLoading: boolean = false;
  hasStudentResultsLoadingFailed: boolean = false;

  constructor(private route: ActivatedRoute,
              private studentProfileService: StudentProfileService,
              private feedbackSessionsService: FeedbackSessionsService,
              private studentService: StudentService,
              private instructorService: InstructorService,
              private commentsToCommentTableModel: CommentsToCommentTableModelPipe,
              tableComparatorService: TableComparatorService,
              statusMessageService: StatusMessageService,
              commentService: FeedbackResponseCommentService,
              commentToCommentRowModel: CommentToCommentRowModelPipe) {
    super(commentToCommentRowModel, commentService, statusMessageService, tableComparatorService);
  }

  ngOnInit(): void {
    this.route.queryParams.subscribe((queryParams: any) => {
      this.courseId = queryParams.courseid;
      this.studentEmail = queryParams.studentemail;

      this.loadStudentRecords();
      this.loadStudentResults();
      this.photoUrl
          = `${environment.backendUrl}/webapi/student/profilePic?`
            + `courseid=${this.courseId}&studentemail=${this.studentEmail}`;
      this.instructorService.getInstructor({
        courseId: queryParams.courseid,
        intent: Intent.FULL_DETAIL,
      }).subscribe((instructor: Instructor) => {
        this.currInstructorName = instructor.name;
      });
    }, (resp: ErrorMessageOutput) => {
      this.statusMessageService.showErrorToast(resp.error.message);
    });
  }

  /**
   * Loads the student's records based on the given course ID and email.
   */
  loadStudentRecords(): void {
    this.hasStudentLoadingFailed = false;
    this.hasStudentProfileLoadingFailed = false;
    this.isStudentLoading = true;
    this.isStudentProfileLoading = true;
    this.studentService.getStudent(
        this.courseId, this.studentEmail,
    ).pipe(finalize(() => this.isStudentLoading = false)).subscribe((resp: Student) => {
      this.studentName = resp.name;
      this.studentTeam = resp.teamName;
      this.studentSection = resp.sectionName;
    }, (resp: ErrorMessageOutput) => {
      this.hasStudentLoadingFailed = true;
      this.statusMessageService.showErrorToast(resp.error.message);
    });
    this.studentProfileService.getStudentProfile(
        this.studentEmail, this.courseId,
    ).pipe(finalize(() => this.isStudentProfileLoading = false)).subscribe((resp: StudentProfile) => {
      this.studentProfile = resp;
    }, (resp: ErrorMessageOutput) => {
      this.hasStudentProfileLoadingFailed = true;
      this.statusMessageService.showErrorToast(resp.error.message);
    });
  }

  /**
   * Loads the student's feedback session results based on the given course ID and student name.
   */
  loadStudentResults(): void {
    this.sessionTabs = [];
    this.hasStudentResultsLoadingFailed = false;
    this.isStudentResultsLoading = true;
    this.feedbackSessionsService.getFeedbackSessionsForInstructor(this.courseId).pipe(
        mergeMap((feedbackSessions: FeedbackSessions) => feedbackSessions.feedbackSessions),
        mergeMap((feedbackSession: FeedbackSession) => {
          return this.feedbackSessionsService.getFeedbackSessionResults({
            courseId: this.courseId,
            feedbackSessionName: feedbackSession.feedbackSessionName,
            groupBySection: this.studentSection,
            intent: Intent.FULL_DETAIL,
          }).pipe(map((results: SessionResults) => {
            // sort questions by question number
            results.questions.sort((a: QuestionOutput, b: QuestionOutput) =>
                a.feedbackQuestion.questionNumber - b.feedbackQuestion.questionNumber);
            return { results, feedbackSession };
          }));
        }),
        finalize(() => this.isStudentResultsLoading = false),
    ).subscribe(
        ({ results, feedbackSession }: { results: SessionResults, feedbackSession: FeedbackSession }) => {
          const giverQuestions: QuestionOutput[] = JSON.parse(JSON.stringify(results.questions));
          giverQuestions.forEach((questions: QuestionOutput) => {
            questions.allResponses = questions.allResponses.filter((response: ResponseOutput) =>
                !response.isMissingResponse && response.giverEmail === this.studentEmail);
          });
          const responsesGivenByStudent: QuestionOutput[] =
              giverQuestions.filter((questions: QuestionOutput) => questions.allResponses.length > 0);

          const recipientQuestions: QuestionOutput[] = JSON.parse(JSON.stringify(results.questions));
          recipientQuestions.forEach((questions: QuestionOutput) => {
            questions.allResponses = questions.allResponses.filter((response: ResponseOutput) =>
                !response.isMissingResponse && response.recipientEmail === this.studentEmail);
          });
          const responsesReceivedByStudent: QuestionOutput[] =
              recipientQuestions.filter((questions: QuestionOutput) => questions.allResponses.length > 0);

          this.sessionTabs.push({
            feedbackSession,
            responsesGivenByStudent,
            responsesReceivedByStudent,
            isCollapsed: false,
          });
          results.questions.forEach((questions: QuestionOutput) => this.preprocessComments(questions.allResponses));
        }, (errorMessageOutput: ErrorMessageOutput) => {
          this.hasStudentResultsLoadingFailed = true;
          this.statusMessageService.showErrorToast(errorMessageOutput.error.message);
        }, () => this.sortFeedbackSessions());
  }

  /**
   * Preprocesses the comments from instructor.
   *
   * <p>The instructor comment will be moved to map {@code instructorCommentTableModel}. The original
   * instructor comments associated with the response will be deleted.
   */
  preprocessComments(responses: ResponseOutput[]): void {
    const timezone: string = this.sessionTabs[0] != null ? this.sessionTabs[0].feedbackSession.timeZone : '';
    responses.forEach((response: ResponseOutput) => {
      this.instructorCommentTableModel[response.responseId] =
          this.commentsToCommentTableModel.transform(response.instructorComments, false, timezone);
      this.sortComments(this.instructorCommentTableModel[response.responseId]);
      // clear the original comments for safe as instructorCommentTableModel will become the single point of truth
      response.instructorComments = [];
    });
  }

  /**
   * Sorts the student's feedback sessions according to name
   */
  private sortFeedbackSessions(): void {
    this.sessionTabs.sort((a: SessionTab, b: SessionTab) => {
      return this.tableComparatorService.compare(
          SortBy.SESSION_NAME,
          SortOrder.ASC,
          a.feedbackSession.feedbackSessionName,
          b.feedbackSession.feedbackSessionName);
    });
  }
}
