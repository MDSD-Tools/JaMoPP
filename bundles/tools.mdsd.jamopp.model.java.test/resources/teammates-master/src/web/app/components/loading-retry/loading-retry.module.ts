import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { LoadingRetryComponent } from './loading-retry.component';

/**
 * Module for the retry button
 */
@NgModule({
  declarations: [LoadingRetryComponent],
  imports: [
    CommonModule,
  ],
  exports: [LoadingRetryComponent],
})
export class LoadingRetryModule { }
