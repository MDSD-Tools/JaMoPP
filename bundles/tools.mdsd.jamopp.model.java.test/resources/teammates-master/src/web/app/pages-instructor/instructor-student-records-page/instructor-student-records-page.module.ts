import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule, Routes } from '@angular/router';
import { NgbCollapseModule } from '@ng-bootstrap/ng-bootstrap';
import { CommentToCommentRowModelPipe } from '../../components/comment-box/comment-to-comment-row-model.pipe';
import { CommentsToCommentTableModelPipe } from '../../components/comment-box/comments-to-comment-table-model.pipe';
import { LoadingRetryModule } from '../../components/loading-retry/loading-retry.module';
import { LoadingSpinnerModule } from '../../components/loading-spinner/loading-spinner.module';
import { PanelChevronModule } from '../../components/panel-chevron/panel-chevron.module';
import {
    GrqRgqViewResponsesModule,
} from '../../components/question-responses/grq-rgq-view-responses/grq-rgq-view-responses.module';
import { StudentProfileModule } from '../student-profile/student-profile.module';
import { InstructorStudentRecordsPageComponent } from './instructor-student-records-page.component';

const routes: Routes = [
  {
    path: '',
    component: InstructorStudentRecordsPageComponent,
  },
];

/**
 * Module for instructor student records page.
 */
@NgModule({
  declarations: [
    InstructorStudentRecordsPageComponent,
  ],
  exports: [
    InstructorStudentRecordsPageComponent,
  ],
  imports: [
    CommonModule,
    StudentProfileModule,
    NgbCollapseModule,
    GrqRgqViewResponsesModule,
    RouterModule.forChild(routes),
    LoadingSpinnerModule,
    LoadingRetryModule,
    PanelChevronModule,
  ],
  providers: [
    CommentToCommentRowModelPipe,
    CommentsToCommentTableModelPipe,
  ],
})
export class InstructorStudentRecordsPageModule { }
