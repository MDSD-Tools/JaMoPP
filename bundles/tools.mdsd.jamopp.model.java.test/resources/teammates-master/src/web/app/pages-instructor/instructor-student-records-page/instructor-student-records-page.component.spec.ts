import { HttpClientTestingModule } from '@angular/common/http/testing';
import { Component, Input } from '@angular/core';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { ActivatedRoute } from '@angular/router';
import { RouterTestingModule } from '@angular/router/testing';
import { NgbModule } from '@ng-bootstrap/ng-bootstrap';
import { of } from 'rxjs';
import { Gender, StudentProfile } from '../../../types/api-output';
import { CommentsToCommentTableModelPipe } from '../../components/comment-box/comments-to-comment-table-model.pipe';
import { LoadingRetryModule } from '../../components/loading-retry/loading-retry.module';
import { LoadingSpinnerModule } from '../../components/loading-spinner/loading-spinner.module';
import { PanelChevronModule } from '../../components/panel-chevron/panel-chevron.module';
import {
  GrqRgqViewResponsesModule,
} from '../../components/question-responses/grq-rgq-view-responses/grq-rgq-view-responses.module';
import { InstructorStudentRecordsPageComponent } from './instructor-student-records-page.component';

@Component({ selector: 'tm-student-profile', template: '' })
class StudentProfileStubComponent {
  @Input() studentProfile: StudentProfile | undefined;
  @Input() studentName: string = '';
  @Input() photoUrl: string = '/assets/images/profile_picture_default.png';
  @Input() hideMoreInfo: boolean = false;
}
@Component({ selector: 'tm-more-info', template: '' })
class MoreInfoStubComponent {
  @Input() studentName: string = '';
  @Input() moreInfoText: string = '';
}

describe('InstructorStudentRecordsPageComponent', () => {
  let component: InstructorStudentRecordsPageComponent;
  let fixture: ComponentFixture<InstructorStudentRecordsPageComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [
        InstructorStudentRecordsPageComponent,
        StudentProfileStubComponent,
        MoreInfoStubComponent,
      ],
      imports: [
        HttpClientTestingModule,
        RouterTestingModule,
        NgbModule,
        GrqRgqViewResponsesModule,
        LoadingSpinnerModule,
        LoadingRetryModule,
        PanelChevronModule,
      ],
      providers: [
        {
          provide: ActivatedRoute,
          useValue: {
            queryParams: of({ courseid: 'su1337', studentemail: 'punk@punk.com' }),
          },
        },
        CommentsToCommentTableModelPipe,
      ],
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(InstructorStudentRecordsPageComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should snap with default fields', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('should snap with populated fields', () => {
    const studentProfile: StudentProfile = {
      name: 'John Doe',
      shortName: 'JD',
      email: 'jd@jd.com',
      institute: 'Area51',
      nationality: 'Antarctican',
      gender: Gender.OTHER,
      moreInfo: '',
    };

    component.studentName = 'Not John Doe';
    component.studentProfile = studentProfile;
    component.courseId = 'su1337';
    component.isStudentLoading = false;
    component.isStudentProfileLoading = false;
    component.isStudentResultsLoading = false;
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should snap when student results are still loading', () => {
    const studentProfile: StudentProfile = {
      name: 'John Doe',
      shortName: 'JD',
      email: 'jd@jd.com',
      institute: 'Area51',
      nationality: 'Antarctican',
      gender: Gender.OTHER,
      moreInfo: '',
    };
    component.studentName = 'John Doe';
    component.studentProfile = studentProfile;
    component.courseId = 'CS1111';
    component.isStudentResultsLoading = true;
    component.isStudentLoading = false;
    component.isStudentProfileLoading = false;

    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should output a correctly formatted url string', () => {
    expect(component.photoUrl)
      .toEqual('http://localhost:8080/webapi/student/profilePic?courseid=su1337&studentemail=punk@punk.com');
  });
});
