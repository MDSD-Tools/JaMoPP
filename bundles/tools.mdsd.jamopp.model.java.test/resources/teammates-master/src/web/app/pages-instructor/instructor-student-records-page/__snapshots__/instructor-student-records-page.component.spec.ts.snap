// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstructorStudentRecordsPageComponent should snap when student results are still loading 1`] = `
<tm-instructor-student-records-page
  commentService={[Function FeedbackResponseCommentService]}
  commentToCommentRowModel={[Function CommentToCommentRowModelPipe]}
  commentsToCommentTableModel={[Function CommentsToCommentTableModelPipe]}
  courseId={[Function String]}
  feedbackSessionsService={[Function FeedbackSessionsService]}
  hasStudentLoadingFailed="false"
  hasStudentProfileLoadingFailed="false"
  hasStudentResultsLoadingFailed="false"
  instructorCommentTableModel={[Function Object]}
  instructorService={[Function InstructorService]}
  isStudentLoading="false"
  isStudentProfileLoading="false"
  isStudentResultsLoading={[Function Boolean]}
  photoUrl={[Function String]}
  route={[Function Object]}
  sessionTabs={[Function Array]}
  statusMessageService={[Function StatusMessageService]}
  studentEmail={[Function String]}
  studentName={[Function String]}
  studentProfile={[Function Object]}
  studentProfileService={[Function StudentProfileService]}
  studentSection=""
  studentService={[Function StudentService]}
  studentTeam=""
  tableComparatorService={[Function TableComparatorService]}
>
  <h1
    id="records-header"
  >
    John Doe's Records
    <small
      class="text-muted text-break"
    >
       - CS1111
    </small>
  </h1><tm-loading-retry>
    
    
    <tm-student-profile />
    
    <tm-more-info />
    
    
  </tm-loading-retry><h2>
    Records in feedback sessions
  </h2><tm-loading-retry>
    
    
    <tm-loading-spinner>
      <div
        class="loading-container"
      >
        <div
          class="row spinner-border text-primary"
          role="status"
        />
        <div
          class="row text-secondary"
        >
          Loading...
        </div>
      </div>
    </tm-loading-spinner>
    
    
  </tm-loading-retry>
</tm-instructor-student-records-page>
`;

exports[`InstructorStudentRecordsPageComponent should snap with default fields 1`] = `
<tm-instructor-student-records-page
  commentService={[Function FeedbackResponseCommentService]}
  commentToCommentRowModel={[Function CommentToCommentRowModelPipe]}
  commentsToCommentTableModel={[Function CommentsToCommentTableModelPipe]}
  courseId={[Function String]}
  feedbackSessionsService={[Function FeedbackSessionsService]}
  hasStudentLoadingFailed="false"
  hasStudentProfileLoadingFailed="false"
  hasStudentResultsLoadingFailed="false"
  instructorCommentTableModel={[Function Object]}
  instructorService={[Function InstructorService]}
  isStudentLoading={[Function Boolean]}
  isStudentProfileLoading={[Function Boolean]}
  isStudentResultsLoading={[Function Boolean]}
  photoUrl={[Function String]}
  route={[Function Object]}
  sessionTabs={[Function Array]}
  statusMessageService={[Function StatusMessageService]}
  studentEmail={[Function String]}
  studentName=""
  studentProfile={[Function Object]}
  studentProfileService={[Function StudentProfileService]}
  studentSection=""
  studentService={[Function StudentService]}
  studentTeam=""
  tableComparatorService={[Function TableComparatorService]}
>
  <h1
    id="records-header"
  >
    's Records
    <small
      class="text-muted text-break"
    >
       - su1337
    </small>
  </h1><tm-loading-retry>
    
    
    <tm-loading-spinner>
      <div
        class="loading-container"
      >
        <div
          class="row spinner-border text-primary"
          role="status"
        />
        <div
          class="row text-secondary"
        >
          Loading...
        </div>
      </div>
    </tm-loading-spinner>
    
    <tm-loading-spinner>
      <div
        class="loading-container"
      >
        <div
          class="row spinner-border text-primary"
          role="status"
        />
        <div
          class="row text-secondary"
        >
          Loading...
        </div>
      </div>
    </tm-loading-spinner>
    
    
  </tm-loading-retry><h2>
    Records in feedback sessions
  </h2><tm-loading-retry>
    
    
    <tm-loading-spinner>
      <div
        class="loading-container"
      >
        <div
          class="row spinner-border text-primary"
          role="status"
        />
        <div
          class="row text-secondary"
        >
          Loading...
        </div>
      </div>
    </tm-loading-spinner>
    
    
  </tm-loading-retry>
</tm-instructor-student-records-page>
`;

exports[`InstructorStudentRecordsPageComponent should snap with populated fields 1`] = `
<tm-instructor-student-records-page
  commentService={[Function FeedbackResponseCommentService]}
  commentToCommentRowModel={[Function CommentToCommentRowModelPipe]}
  commentsToCommentTableModel={[Function CommentsToCommentTableModelPipe]}
  courseId={[Function String]}
  feedbackSessionsService={[Function FeedbackSessionsService]}
  hasStudentLoadingFailed="false"
  hasStudentProfileLoadingFailed="false"
  hasStudentResultsLoadingFailed="false"
  instructorCommentTableModel={[Function Object]}
  instructorService={[Function InstructorService]}
  isStudentLoading="false"
  isStudentProfileLoading="false"
  isStudentResultsLoading="false"
  photoUrl={[Function String]}
  route={[Function Object]}
  sessionTabs={[Function Array]}
  statusMessageService={[Function StatusMessageService]}
  studentEmail={[Function String]}
  studentName={[Function String]}
  studentProfile={[Function Object]}
  studentProfileService={[Function StudentProfileService]}
  studentSection=""
  studentService={[Function StudentService]}
  studentTeam=""
  tableComparatorService={[Function TableComparatorService]}
>
  <h1
    id="records-header"
  >
    Not John Doe's Records
    <small
      class="text-muted text-break"
    >
       - su1337
    </small>
  </h1><tm-loading-retry>
    
    
    <tm-student-profile />
    
    <tm-more-info />
    
    
  </tm-loading-retry><h2>
    Records in feedback sessions
  </h2><tm-loading-retry>
    
    
    <div>
      
    </div>
    
    
  </tm-loading-retry>
</tm-instructor-student-records-page>
`;
